### PR TITLE
read me modifications

### PR DESCRIPTION
**Pull request by AnaisBellia**
modification du read me

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -248,8 +248,15 @@ QAndQDotBounds waits for a biorbd model and returns a structure with the minimal
 How convenient!
 ```python
 x_bounds = QAndQDotBounds(biorbd_model)
-```
-Then, override the first and last column to be 0, that is the sideways and rotation to be null for both the position and the velocities
+
+
```
Comment (pariterre) [inline README.md:252]: Is there a reason for the extra spacing here?

File: README.md [hunk 2]
```diff
@@ -280,8 +287,13 @@ Still, helping the solver is usually a good idea, so let's give ̀`Ipopt` a star
 The initial guess that we can provide are those for the states (`x_init`, here *q* and *qdot*) and for the controls (`u_init`, here *tau*). 
 So let's define both of them quickly
 ```python
-x_init = InitialGuess([0, 0, 0, 0])
+x_init = InitialGuess([0, 0, 0, 0])
```
Comment (pariterre) [inline README.md:290]: Why did you add a space?

File: README.md [hunk 3]
```diff
@@ -248,8 +248,15 @@ QAndQDotBounds waits for a biorbd model and returns a structure with the minimal
 How convenient!
 ```python
 x_bounds = QAndQDotBounds(biorbd_model)
-```
-Then, override the first and last column to be 0, that is the sideways and rotation to be null for both the position and the velocities
+
+
+The first element of x_bounds is the degree of freedom declared in order in the bio.Mod file.
```
Comment (pariterre) [inline README.md:253]: first elements ... are ... as declared? (plural)
Please note that this is not always the case as a user can change the order as he is pleased... So I am not sure it make sense to write this here.. Maybe specify that it is the default settings for TORQUE_DRIVEN dynamics?

`in order`? `in the same order as in`?

Comment (pariterre) [reply README.md:253]: Did you mean .bioMod?

Comment (pariterre) [reply README.md:253]: Also, please note that the exact same idea is said litteraly two sentences before (`To define that, it would be nice to first define boundary constraints on the position (*q*) and velocities (*qdot*) that match those in the bioMod file and to apply them at the very beginning, the very end and all the intermediate nodes as well.`)
Maybe update this sentence instead of producing a new one

File: README.md [hunk 4]
```diff
@@ -248,8 +248,15 @@ QAndQDotBounds waits for a biorbd model and returns a structure with the minimal
...
+In this case, the degree of freedom with index 0 is translation y, and the index 1 refers to rotation x. 
+Finally, the index 2 and 3 are respectively the velocity of translation y and rotation x
```
Comment (pariterre) [inline README.md:255]: This would make more sense after the sentence (The pendulum is required to start in a downward position (0 rad) and to finish in an upward position (3.14 rad) with no velocity at start and end nodes.)

File: README.md [hunk 5]
```diff
@@ -248,8 +248,15 @@ QAndQDotBounds waits for a biorbd model and returns a structure with the minimal
...
 x_bounds = QAndQDotBounds(biorbd_model)
-```
-Then, override the first and last column to be 0, that is the sideways and rotation to be null for both the position and the velocities
+
+
+The first element of x_bounds is the degree of freedom declared in order in the bio.Mod file.
+In this case, the degree of freedom with index 0 is translation y, and the index 1 refers to rotation x. 
+Finally, the index 2 and 3 are respectively the velocity of translation y and rotation x 
+
+The time is discretized in nodes wich is the second element declared in x_bounds
```
Comment (pariterre) [inline README.md:257]: what are you calling "element"? I though you were referring to a subset of the rows (as the full rows are not just *q* based as you seem to imply here: `The first element of x_bounds is the degree of freedom declared in order in the bio.Mod file.`, but to the full path constraints of the problem (including *qdot*). But now, `element` seems to refer to dimensions (i.e. the columns), because you refer to time. If so, please use `dimension` and not element. Also, make sure you don't mislead the user by saying the rows refer to *q* solely as it is not. 

Finally, the columns in x_bounds can be discretized the same way as the problem, but this is not even the default option. The default is 3 columns, that is the  `starting node`, the `intermediate nodes` and the `final node`

File: README.md [hunk 6]
```diff
@@ -248,8 +248,15 @@ QAndQDotBounds waits for a biorbd model and returns a structure with the minimal
...
+
+We want the first and last column(which is equivalent to nodes 0 and -1) to be 0, that is the translations and rotations to be null for both the position and so the velocities, wich finally means that all the degrees of freedom are null.
```
Comment (pariterre) [inline README.md:259]: which*

degrees of freedom or *q* do not refer to both position and velocities, but only to position. So this final part (from which to the end) is misleading.

File: README.md [hunk 7]
```diff
@@ -280,8 +287,13 @@ Still, helping the solver is usually a good idea, so let's give ̀`Ipopt` a star
 The initial guess that we can provide are those for the states (`x_init`, here *q* and *qdot*) and for the controls (`u_init`, here *tau*). 
 So let's define both of them quickly
 ```python
-x_init = InitialGuess([0, 0, 0, 0])
+x_init = InitialGuess([0, 0, 0, 0]) 
 u_init = InitialGuess([0, 0])
+
+Each element declared in x_init is the value of the degree of freedom in the bio.Mod file :
```
Comment (pariterre) [inline README.md:293]: This is not true, see prior comments

Comment (pariterre) [reply README.md:293]: .bioMod?

File: README.md [hunk 8]
```diff
@@ -280,8 +287,13 @@ Still, helping the solver is usually a good idea, so let's give ̀`Ipopt` a star
 The initial guess that we can provide are those for the states (`x_init`, here *q* and *qdot*) and for the controls (`u_init`, here *tau*). 
 So let's define both of them quickly
 ```python
-x_init = InitialGuess([0, 0, 0, 0])
+x_init = InitialGuess([0, 0, 0, 0]) 
 u_init = InitialGuess([0, 0])
+
+Each element declared in x_init is the value of the degree of freedom in the bio.Mod file : 
+for example x_init = InitialGuess([q1, q2, qdot1, qdot2]) with q1,q2 the positions and qdot1,qdot2 the velocities of q1 and q2
```
Comment (pariterre) [inline README.md:294]: For?

please put codes and keywords inside the "accent grave" brackets (as in à without the a under :P) and q1 and so on inside star to make them italic

File: README.md [hunk 9]
```diff
@@ -280,8 +287,13 @@ Still, helping the solver is usually a good idea, so let's give ̀`Ipopt` a star
 The initial guess that we can provide are those for the states (`x_init`, here *q* and *qdot*) and for the controls (`u_init`, here *tau*). 
 So let's define both of them quickly
 ```python
-x_init = InitialGuess([0, 0, 0, 0])
+x_init = InitialGuess([0, 0, 0, 0]) 
 u_init = InitialGuess([0, 0])
+
+Each element declared in x_init is the value of the degree of freedom in the bio.Mod file : 
+for example x_init = InitialGuess([q1, q2, qdot1, qdot2]) with q1,q2 the positions and qdot1,qdot2 the velocities of q1 and q2 
+We have both the positions and their velocities to be 0 :
```
Comment (pariterre) [inline README.md:295]: Why the colon?

File: README.md [hunk 10]
```diff
@@ -650,7 +662,7 @@ Since this is an Enum, it is possible to use tab key on the keyboard to dynamica
 Please note that one can change the dynamic function associated to any of the configuration by providing a custom dynamics_function. 
 For more information on this, please refer to the Dynamics and DynamicsList section right before. 
 
-#### TORQUE_DRIVEN
+#### TORQUE_DRIVEN
```
Comment (pariterre) [inline README.md:665]: Why did you add a space here?

File: README.md [hunk 11]
```diff
@@ -880,7 +892,7 @@ constraint_list = ConstraintList()
 constraint_list.add(constraint)
 ```
 
-### Class: ConstraintFcn
+### Class: ConstraintFcn (pour TRACK_STATE, TRACK_MARKERS ect donner quelques exemples de parametres, expliciter)
```
Comment (pariterre) [inline README.md:895]: Is this intended?

File: README.md (hunk 1 above)
Comment (AnaisBellia) [reply README.md:252]: espaces supprimés

File: README.md (hunk 3 above)
Comment (AnaisBellia) [reply README.md:253]: "In the default settings for TORQUE_DRIVEN dynamics, the first elements of x_bounds are the degrees of freedom  (*q*)  and their velocities (*qdot*) that match those in the bioMod file. The time is discretized in nodes wich are the second elements declared in x_bounds
If you have more than one phase, we shall have x_bound[phase][degrees of freedom, nodes]"

File: README.md (hunk 11 above)
Comment (AnaisBellia) [reply README.md:895]: On souhaitait rajouter tous les paramètres que l'utilisateur peut rajouter tels que "min_bound" "phase"

File: README.md (hunk 5 above)
Comment (AnaisBellia) [reply README.md:257]: mot "element" remplacé par "dimension". Nous voulions parler de dimension en effet pour expliquer que les premières valeurs rentrées dans les crochets de x_bounds étaient les degrés de liberté q et leur vitesse qdot puis les noeuds.

File: README.md [hunk 12]
```diff
@@ -243,20 +243,17 @@ dynamics = Dynamics(DynamicsFcn.TORQUE_DRIVEN)
 ```
 
 The pendulum is required to start in a downward position (0 rad) and to finish in an upward position (3.14 rad) with no velocity at start and end nodes.
-To define that, it would be nice to first define boundary constraints on the position (*q*) and velocities (*qdot*) that match those in the bioMod file and to apply them at the very beginning, the very end and all the intermediate nodes as well.
+In this case, the degree of freedom with index 0 is translation y, and the index 1 refers to rotation x.
```
Comment (pariterre) [inline README.md:246]: rotation about x

Comment (pariterre) [reply README.md:246]: The `state` instead of degree of freedom, as indices refer to the state variable (where the first part is the degree and the second is obviously not!)

File: README.md [hunk 13]
```diff
@@ -243,20 +243,17 @@ dynamics = Dynamics(DynamicsFcn.TORQUE_DRIVEN)
...
-To define that, it would be nice to first define boundary constraints on the position (*q*) and velocities (*qdot*) that match those in the bioMod file and to apply them at the very beginning, the very end and all the intermediate nodes as well.
+In this case, the degree of freedom with index 0 is translation y, and the index 1 refers to rotation x. 
+Finally, the index 2 and 3 are respectively the velocity of translation y and rotation x 
+
 QAndQDotBounds waits for a biorbd model and returns a structure with the minimal and maximal bounds for all the degrees of freedom and velocities on three columns corresponding to the starting node, the intermediate nodes and the final node, respectively.
 How convenient!
 ```python
 x_bounds = QAndQDotBounds(biorbd_model)
+The first dimension of x_bounds is the degrees of freedom (*q*) `and` their velocities (*qdot*) that match those `in` the bioMod `file`. The time `is` discretized `in` nodes wich `is` the second dimension declared `in` x_bounds.
+If you have more than one phase, we would have x_bound[*phase*][*q `and` qdot*, *nodes*]
```
Comment (pariterre) [inline README.md:254]: It seems that you remove the closing ```?

File: README.md [hunk 14]
```diff
@@ -282,8 +286,10 @@ So let's define both of them quickly
 ```python
 x_init = InitialGuess([0, 0, 0, 0])
 u_init = InitialGuess([0, 0])
+
 ```
 Please note that `x_init` is twice the size of `u_init` because it contains the two degrees of freedom from the generalized coordinates (*q*) and the two from the generalized velocities (*qdot*), while `u_init` only contains the generalized forces (*tau*)
+In this case, we have both the positions `and` their velocities to be 0
```
Comment (pariterre) [inline README.md:292]: It seems that there is a full stop missing in the last sentence

File: README.md [hunk 15]
```diff
@@ -243,16 +243,16 @@ dynamics = Dynamics(DynamicsFcn.TORQUE_DRIVEN)
...
+Finally, the index 2 and 3 are respectively the velocity of translation y and rotation about x 
 
 QAndQDotBounds waits for a biorbd model and returns a structure with the minimal and maximal bounds for all the degrees of freedom and velocities on three columns corresponding to the starting node, the intermediate nodes and the final node, respectively.
 How convenient!
 ```python
 x_bounds = QAndQDotBounds(biorbd_model)
 The first dimension of x_bounds is the degrees of freedom (*q*) `and` their velocities (*qdot*) that match those `in` the bioMod `file`. The time `is` discretized `in` nodes wich `is` the second dimension declared `in` x_bounds.
 If you have more than one phase, we would have x_bound[*phase*][*q `and` qdot*, *nodes*]
-
+```
```
Comment (pariterre) [inline README.md:255]: This goes before the paragraph:
```python
x_bounds = QAndQDotBounds(biorbd_model)
x_bounds = QAndQDotBounds(biorbd_model)
```

File: README.md [hunk 16]
```diff
@@ -289,7 +289,7 @@ u_init = InitialGuess([0, 0])
 
 ```
 Please note that `x_init` is twice the size of `u_init` because it contains the two degrees of freedom from the generalized coordinates (*q*) and the two from the generalized velocities (*qdot*), while `u_init` only contains the generalized forces (*tau*)
-In this case, we have both the positions `and` their velocities to be 0
+In this case, we have both the positions `and` their velocities to be 0.
```
Comment (pariterre) [inline README.md:292]: And the previous sentence too ;)

**Review by pariterre — CHANGES_REQUESTED** [review]
Two small changes and I accept :)
